annnufan/assignment1/solution_1.cpp

```diff
@@ -0,0 +1,104 @@
+#include <iostream>
+#include <string>
+#include <cstring>
+#include <vector>
+#include <algorithm>
+#include <cctype>
+#include <functional>
+
+void word_anagram(bool case_sens) {
+	std::string word_1, word_2;
+	std::cin >> word_1 >> word_2;
+	if (!case_sens) {
+		std::transform(word_1.begin(), word_1.end(), word_1.begin(), std::ptr_fun <int, int> (std::tolower));
+		std::transform(word_2.begin(), word_2.end(), word_2.begin(), std::ptr_fun <int, int> (std::tolower));
+	}
+	if (word_1.length() != word_2.length()) {
+		std::cout << "Second word isn't anagram of first\n";
+		return;
+	}
+
+	std::vector<int> count(256, 0);
+	for (auto ch : word_1) {
+		count[ch]++;
+	}
+	for (auto ch : word_2) {
+		if (count[ch] > 0) {
+			count[ch]--;
+		} else {
+			std::cout << "Second word isn't anagram of first\n";
+			return;
+		}
+	}
+
+	std::cout << "Yes, second word is anagram of first\n";
+	return;
+}
+
+std::vector<std::string> line_to_words(std::string line) {

```

Comment:
This function does two things:
(1) Splits lines to words.
(2) Sorts each word.

(2) is not reflected in the functions name, and a of this function can be surprised that the words are sorted.
What can we do?

We can add comments to reflect that:
```C++
// Each word in the result vector is alphabetically sorted.
```
Better than a comment, we can choose a name that reflects that:
```C++
std::vector<std::string> line_to_sorted_words(std::string line) {
```
But even better is to make this function do only one thing and name it accordingly.
This function can *only* split a line to words, and only later you can go over the result and sort each word.
This way this function can be better reused by a different user which only wants to split a sentence (very common use) without sorting the words (which is not very common).